api/v1/resourceset_types.go

```diff
@@ -57,6 +63,22 @@ type ResourceSetSpec struct {
 	Wait bool `json:"wait,omitempty"`
 }
 
+type InputsProvider struct {
+	// APIVersion of the inputs provider resource.
+	// When not set, the APIVersion of the ResourceSet is used.
+	// +optional
+	APIVersion string `json:"apiVersion,omitempty"`
+
+	// Kind of the inputs provider resource.
+	// +kubebuilder:validation:Enum=ResourceSetInputProvider
+	// +required
+	Kind string `json:"kind"`
```

Comment:
If we plan to support multiple object types here, shouldn't we define an interface `interface{ GetInputs() ([]map[string]any, error) }` and implement it for `ResourceSetInputProvider`? (Similar to source-controller's `Source` interface containing `GetArtifact() *Artifact`)

Reply:
Other kinds (Kubernetes ConfigMaps and Secrets) will not subscribe to this interface. I'm also not for premature optimisation, since now we only support one type, there is no need to introduce a interface.

Reply:
But we already do support two types here: `*ResourceSet`, which provides static inputs, and `*ResourceSetInputProvider`, which provides dynamic inputs. See the next comment below.

Reply:
Also, when we implement ConfigMaps and Secrets in another PR, we could easily have simple wrappers on top of those types to implement the interface:

```go
type configMapInputProvider struct { *corev1.ConfigMap }

func (in configMapInputProvider) GetInputs() ([]map[string]any, error)
```